src/core/mesh/qgsmeshlayertemporalproperties.h

```diff
@@ -0,0 +1,72 @@
+#ifndef QGSMESHLAYERTEMPORALPROPERTIES_H
+#define QGSMESHLAYERTEMPORALPROPERTIES_H
+
+#include "qgsmaplayertemporalproperties.h"
+
+
+/**
+ * \class QgsMeshLayerTemporalProperties
+ * \ingroup core
+ * Implementation of map layer temporal properties for mesh layers.
+ *
+ *
+ * The time in a mesh layer is defined by :
+ * - a reference time provided by the data, the project or the user
+ * - each dataset is associated with a relative times
+ * - time extent is defined by the first time and the last time of all dataset
+ *
+ * Reference time :          AT
+ * Dataset 1 time            o-----RT------RT-----RT-----------RT
+ * Dataset 2 time            o---------RT------RT--------RT
+ * Dataset 3 time            o------------------------------RT-------RT----------RT
+ * Time extent of layer      o-----<--------------------------------------------->
+ *
+ * AT : absolute time (QDateTime)
+ * RT : relative time (qint64)
+ *
+ * \since QGIS 3.14
+ */
+
+class CORE_EXPORT QgsMeshLayerTemporalProperties : public QgsMapLayerTemporalProperties
```

Comment:
what is the difference between QgsMeshLayerTemporalProperties and QgsMeshLayerTemporalCapabilities?

Reply:
I think the principle of new temporal classes (capabilities and properties) is :

    capabilities : what can the provider handle from the data, attribute of the class are only set from the provider and from data, as much as possible (provider time unit is a breach in the rule, but it is a provider capability that needs to be used in the capabilities class).
    properties : settings from other part, that is user or project settings.


Those classes was also implemented for raster wms-t layer